src/compose/ComposeBox.js

```diff
@@ -217,6 +219,36 @@ class ComposeBox extends PureComponent<Props, State> {
     }
   };
 
+  replaceMessageTextAtRange = (text: string, beg: number, end: number) => {
+    const { message } = this.state;
+    this.setMessageInputValue(message.slice(0, beg) + text + message.slice(end, message.length));
+    return [beg, beg + text.length];
+  };
+
+  insertAttachment = async (attachments: [DocumentPickerResponse]) => {
+    this.setState(({ isUploading }) => ({
+      isUploading: !isUploading,
+    }));
+
+    const { _, auth } = this.props;
+    let [beg, end] = [undefined, this.state.selection.start];
+    for (let i = 0; i < attachments.length; i++) {
+      const fileName = attachments[i].name ?? `attachment-${i + 1}`;
+      [beg, end] = this.replaceMessageTextAtRange(`[Uploading ${fileName}...]()\n`, end, end);
+      try {
+        const response = await api.uploadFile(auth, attachments[i].uri, fileName);
+        [beg, end] = this.replaceMessageTextAtRange(`[${fileName}](${response.uri})\n\n`, beg, end);
+      } catch (e) {
+        [beg, end] = this.replaceMessageTextAtRange('', beg, end);
+        showToast(_(`Uploading ${fileName} unsuccessful`));
```

Comment:
We'll want the string being translated to be a constant string, with a placeholder in it for the filename. See [docs/howto/translations.md](https://github.com/zulip/zulip-mobile/blob/8430b6da8/docs/howto/translations.md), and search for "interpolating".

We'll also want to add the string to `static/translations/messages_en.json`; from there after the change is merged we'll sync it over to Transifex so that contributors there can translate it. See the next few paragraphs in the same doc.

Reply:
Just a concern regarding this, [docs/howto/translations.md](https://github.com/zulip/zulip-mobile/blob/8430b6da8/docs/howto/translations.md) says -
> Translators will translate the constant string 'Hello, {name}', including the placeholder.

does that mean we also converts the filename variable to another language and if so shouldn't that remain preserved since it indicates the name of the file in user's device?

Reply:
Ah, good question -- that explanation was ambiguous. Does this answer the question?
be39f564a